vector/src/main/java/im/vector/activity/VectorHomeActivity.java

```diff
@@ -1442,6 +1465,202 @@ public void run() {
+
+                View iconView = navigationItemView.findViewById(R.id.icon);
+
+                if (iconView.getParent() instanceof FrameLayout) {
+                    UnreadCounterBadgeView badgeView = new UnreadCounterBadgeView(iconView.getContext());
+
+                    // compute the new position
+                    FrameLayout.LayoutParams iconViewLayoutParams = (FrameLayout.LayoutParams)iconView.getLayoutParams();
+                    FrameLayout.LayoutParams badgeLayoutParams = new FrameLayout.LayoutParams(FrameLayout.LayoutParams.WRAP_CONTENT, FrameLayout.LayoutParams.WRAP_CONTENT);
+                    badgeLayoutParams.setMargins(iconViewLayoutParams.leftMargin +  badgeOffsetX, iconViewLayoutParams.topMargin - badgeOffsetY, iconViewLayoutParams.rightMargin, iconViewLayoutParams.bottomMargin);
+                    badgeLayoutParams.gravity = iconViewLayoutParams.gravity;
+
+                    ((FrameLayout) iconView.getParent()).addView(badgeView, badgeLayoutParams);
+                    mBadgeViewByIndex.put(itemId, badgeView);
+                }
+            } catch (Exception e) {
+                Log.e(LOG_TAG, "## addUnreadBadges failed " + e.getMessage());
+            }
+        }
+
+        refreshUnreadBadges();
+    }
+
+    /**
+     * Refresh the badges
+     */
+    private void refreshUnreadBadges() {
+        Collection<RoomSummary> summaries = mSession.getDataHandler().getStore().getSummaries();
+
+        for(Integer id : mBadgeViewByIndex.keySet()) {
+            UnreadCounterBadgeView badgeView = mBadgeViewByIndex.get(id);
+
+            // compute the badge value and its displays
+            int highlightCount = 0;
+            int notificationCount = 0;
+            boolean mustBeHighlighted = false;
+
+            List<String> filteredRoomIds = null;
+
+            if (id == R.id.bottom_action_favourites) {
+                List<Room> favRooms = mSession.roomsWithTag(RoomTag.ROOM_TAG_FAVOURITE);
+
+                filteredRoomIds = new ArrayList<>();
+
+                for(Room room : favRooms) {
+                    filteredRoomIds.add(room.getRoomId());
+                }
+            } else if ((id == R.id.bottom_action_people)){
+                filteredRoomIds = mSession.getDirectChatRoomIdsList();
+            }
+
+            if ((null == filteredRoomIds) || !filteredRoomIds.isEmpty()) {
+                for(RoomSummary summary : summaries) {
+                    Room childRoom = mSession.getDataHandler().getStore().getRoom(summary.getRoomId());
+
+                    // test if the room is allowed
+                    if ((null != childRoom) && ((null == filteredRoomIds) || filteredRoomIds.contains(summary.getRoomId()))) {
+                        highlightCount += childRoom.getHighlightCount();
+                        notificationCount += childRoom.getNotificationCount();
+                        mustBeHighlighted |= summary.isHighlighted();
+                    }
```

Comment:
Maybe each fragment could provide itself the list of rooms it handles (filteredRoomIds)